### PR TITLE
chore: update Go tooling and packages

### PR DESCRIPTION
**Pull request by nfnt**
Also introduces a workspace file. This is helpful in editors using the 'gopls' language server, as these will now be able to work with multiple Go projects at once.
Furthermore, the `fl-operator` project has been updated to track its dependency on `controller-tools` in `go.mod` using a [tools.go file](https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module).

### CODE REVIEW

File: components/fl-operator/Makefile [hunk 1]
```diff
@@ -37,38 +39,44 @@ help: ## Display this help.
 
 ##@ Development
 
+bin/controller-gen:
+	GOBIN="$(GOBIN)" go install sigs.k8s.io/controller-tools/cmd/controller-gen
+
+bin/setup-envtest:
+	GOBIN="$(GOBIN)" go install sigs.k8s.io/controller-runtime/tools/setup-envtest
```
Comment (GerardoGR) [inline components/fl-operator/Makefile:46]: The reason of not installing these only locally in the project, and instead in the `Dockerfile` was to reduce the possible duplicate downloads of the `controller-gen` and `setup-envtest`. I would prefer to reduce that duplication in the Dockerfile and/or rely that these exist in the OS.

**Review by GerardoGR — CHANGES_REQUESTED** [review]
Before merging this I would like to discuss the comment I left other than that the PR looks good.

File: components/fl-operator/Makefile (hunk 1 above)
Comment (nfnt) [reply components/fl-operator/Makefile:46]: I have a different point-of-view: These tools are closely tied to the `controller-runtime` dependency. I.e. older versions of `controller-gen` will fail with new versions of `controller-runtime` and vice versa. Therefore it is important to ensure that the right version of these tools is in use and that we don't rely on existing installations of this tools in the system `$PATH`. Which makes it essential to track the versions of these tools as part of the project -- ideally in `go.mod` as introduced in this PR -- and install them locally only for this project.

Comment (GerardoGR) [reply components/fl-operator/Makefile:46]: I see. I was not aware that the `controller-runtime` versions were not backwards-compatible. Then it makes sense to have them as you propose.

**Review by GerardoGR — APPROVED** [review]

File: components/fl-operator/Makefile (hunk 1 above)
Comment (nfnt) [reply components/fl-operator/Makefile:46]: And keep in mind, as I'm not sure if that's easily visible from the Makefile, that the binaries are not installed globally but only for the project by overriding `GOBIN` and `PATH` when these tools are needed.

**Pull request merged**